pkg/policy/execute.go

```diff
@@ -122,8 +140,11 @@ func (e *Executor) executePolicy(ctx context.Context, progressUpdate UpdateCallb
 	for _, p := range policy.Policies {
 		if len(selector) == 0 || p.Name == selector[0] {
 			found = true
-			r, err := e.executePolicy(ctx, progressUpdate, req, p, rest)
+			executor := e.with(p.Name)
+			executor.log.Info("starting policy execution")
+			r, err := executor.executePolicy(ctx, executor.progressUpdate, req, p, rest)
```

Comment:
question: why are we passed the progressUpdate from the executor in an executor function?